### PR TITLE
Hiero club, crusher trophy and resonator tweaks

### PR DESCRIPTION
**Pull request by Rukofamicom**
## About The Pull Request
* Chasers are now spawned alongside a cardinal burst instead of overriding it, and they also spawn at the point of the burst instead of needing to travel from the player to reach their target. 
* Chasers spawned by both the hierophant club and kinetic crusher with vortex talisman no longer detonate gibtonite, instead defusing it in its weakest form. All other magic attacks with the Hiero club still detonate gibtonite
* The Hiero club's magic attacks now do only 5 damage when the player is not on lavaland (down from 15)
* Resonators have had their damage halved

## Why It's Good For The Game
* **Resonators** - were simply overlooked at the time of the great cutting in half. They do absurd damage now thanks to not having been cut in half, even exceeding crusher backstabs in damage.  

* **Hiero club reduced magic damage** - The ultimate omni-tool on lavaland. Great for cutting paths, great for combat, and even has the ability to teleport. You know what it doesn't need? The ability to kill crew or antags on the station through walls and windows. The greatly reduced magic damage should substantially hinder its ability to be used against against crew and antagonists alike, though that's not to say someone can't still try.

* **Hiero chaser buffs** - The chasers ranged from underwhelming to downright hazardous to the user thanks to their ability to randomly detonate gibtonite, and this was a complaint I received immediately after adjusting the crusher trophy as well. The changes make them far more likely to deal damage in combat, as well as making the club and trophy much safer to use in fights overall

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

## Chaser on Gibtonite
image
image

## Other magic burst on Gibtonite
image
image

</details>

## Changelog
:cl:
balance: Hierophant club now does reduced damage when the player using it isn't on lavaland. 
balance: Hierophant club chasers are now fired alongside cardinal blasts when off of cooldown, instead of replacing them. They will also begin travel from the center of the blast instead of from the user making them much more useful in combat. 
tweak: Hierophant chasers from all sources will no longer detonate gibtonite, instead defusing it in its weakest form. 
fix: Resonators are no longer forgotten and have had their damage halved just like most of the other lavaland tools. They're no longer capable of rapidly dealing more damage than backstabbing with crushers. 
/:cl:

### CODE REVIEW

**Comment by Tsar-Salat** [comment]
Looks good. 

As a suggestion, perhaps a less punishing Hiero cooldown on space ruins?

Benefits:
- Various mobs powered by Hiero technology already are enemies there 
- the vortex gun tech is there already
- it would similarly incentivize space mining as an alternative to lavaland mining, something I think I've seen you been a proponent of?

**Comment by Rukofamicom** [comment]
> As a suggestion, perhaps a less punishing Hiero cooldown on space ruins?

I took the cooldown penalty away entirely in favor of making it impossible to enable friendly fire - it will function at full force in all locations, but only against non-humanoid targets. This accomplishes roughly the same goal I set out to with less impact on non-validhunting efforts.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
The fact that it cannot damage humans at all is very unintuative and needs to be made very clear

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 1]
```diff
@@ -1128,7 +1128,7 @@
 	var/blast_range = 13 //how long the cardinal blast's walls are
 	var/obj/effect/hierophant/beacon //the associated beacon we teleport to
 	var/teleporting = FALSE //if we ARE teleporting
-	var/friendly_fire_check = FALSE //if the blasts we make will consider our faction against the faction of hit targets
+	var/friendly_fire_check = TRUE //if the blasts we make will consider our faction against the faction of hit targets
```
Comment (PowerfulBacon) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Why even keep this in?

Comment (Rukofamicom) [reply code/modules/mining/lavaland/necropolis_chests.dm]: Admins can still override it if they choose

**Comment by Rukofamicom** [comment]
> The fact that it cannot damage humans at all is very unintuative and needs to be made very clear

Going to be honest, I don't see any reasonable way to resolve this issue without introducing others:
* Make it so the club only works on lavaland -> how the fuck to vortex guns work + why can't it be used for asteroid mining (where the teleport really shines)?
* Make it much weaker so it isn't viable in pvp -> players can't defend themselves adequately from fauna with it.

You can go ahead and close this unless you have some direction in that regard.

**Comment by PowerfulBacon** [comment]
We could just allow it to harm humans but make it really crap at doing so

**Comment by Rukofamicom** [comment]
I can make the damage much weaker away from lavaland, though this still presents some dissonance with Vortex guns. 

I think that might be the best available solution though. The club is just inherently clashes with how we want the game designed, but I really don't want to straight up remove it.

**Comment by Rukofamicom** [comment]
There, updated the initial post as well. 

Changes since conversation:
* Ability to toggle friendly fire reinstated
* Club magic now does 5 damage if player is away from lavaland, with a message in chatbox
* Reverted the cooldown increase on melee bursts, since they're no longer a damage burst issue in pvp

**Review by PowerfulBacon — APPROVED** [review]

**Comment by PowerfulBacon** [comment]
We can see if this works

**Pull request merged**